### PR TITLE
Find the Closest Palindrome (Leetcode) Added problem number 564

### PR DESCRIPTION
**Pull request by vivekvardhan2810**
## Related Issue

Fixed : #1006 

## Description

I had Added the Solution of problem "Find the Closest Palindrome" under Leetcode Solution > 0500-0599 > 0564

## Type of PR

- [x] Feature enhancement
- [x] Documentation update

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by vivekvardhan2810** [comment]
Hi @Ajay-Dhangar , I have added solution you can review it.

And tell me the suggestions.

File: dsa-solutions/lc-solutions/0500-0599/0564- Find the Closest Palindrome.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+Output: "0"
+
+Explanation: 0 and 2 are the closest palindromes but we return the smallest which is 0.
+
+```
+
+### Constraints
+
+- $1 <= n.length <= 18$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0500-0599/0564- Find the Closest Palindrome.md:42]: replace `$1 <= n.length <= 18$` to `$1 \leq n.length \leq 18$`

File: dsa-solutions/lc-solutions/0500-0599/0564- Find the Closest Palindrome.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+- $n$ consists of only digits.
+- $n$ does not have leading zeros.
+- $n$ is representing an integer in the range [1, 10^18 - 1].
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0500-0599/0564- Find the Closest Palindrome.md:45]: replace `10^18` to `$10^18$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
change your code based on suggestions

**Comment by vivekvardhan2810** [comment]
@Ajay-Dhangar resolved and done.

**Pull request merged**